cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -398,6 +403,110 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 	return nil
 }
 
+func (a *StaticAutoscaler) handleQuotaAndStockoutErrors(currentTime time.Time) {
+	currentInstancesPerNodeGroup := a.clusterStateRegistry.GetCloudProviderNodeInstances()
+	previousInstancesPerNodeGroup := a.clusterStateRegistry.GetPreviousCloudProviderNodeInstances()
```

Comment:
I don't feel super strongly about this, but exposing this from clusterstate feels kinda ugly. We could move the whole set difference logic to cloudprovider and only expose something like NodesThatFailedToCreate() (new, old).
To me it seems more consistent with what we already do for things like UnregisteredNodes - build data structure describing state in clusterstate, act on it in core/. WDYT?

Reply:
Did you mean move it to clusterstate?

Reply:
I did mean clusterstate, sorry :(

Reply:
Let's talk